src/pip/_internal/index/package_finder.py

```diff
@@ -695,9 +694,11 @@ def set_prefer_binary(self):
         # type: () -> None
         self._candidate_prefs.prefer_binary = True
 
-    def logged_links_rp(self):
+    def skipped_links_requires_python(self):
         # type: () -> List[str]
-        skips = [skip for skip in self._logged_links_rp]
+        skips = [skip[1] for skip in self._logged_links if 'Requires-Python' in skip[1]]
+        # cleans duplicate package version found from different links
+        skips = list(dict.fromkeys(skips))
         skips.sort()
```

Comment:
These three lines can be simplified to

```suggestion
        skips = sorted({reason for _, reason in self._logged_links if 'Requires-Python' in reason})
```